airbyte-api-server/src/main/java/io/airbyte/api/server/repositories/impl/ConnectionsRepositoryRESTImpl.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.api.server.repositories.impl;
+
+import io.airbyte.api.server.repositories.ConnectionsRepository;
+import io.airbyte.api.server.repositories.clients.ConfigApiClient;
+import io.airbyte.api.server.repositories.clients.SyncDto;
+import io.micronaut.data.annotation.Repository;
+import io.micronaut.http.HttpResponse;
+import java.util.UUID;
+import javax.validation.constraints.NotBlank;
+import lombok.extern.slf4j.Slf4j;
+
+@Repository
+@Slf4j
+public class ConnectionsRepositoryRESTImpl implements ConnectionsRepository {
```

Comment:
Could you explain a bit why we are extended the data repository?  It feels a bit odd to have implementations in a repository that don't talk to the underlying datasource, but instead issue an HTTP request.  I would suggest separating these two concerns:  Marking the `ConnectionsRepository` interface with the `@Repository` annotation and then making this class be a wrapper around all of the operations.  In other words, inject both the `ConfigApiClient` and `ConfigRepository` to this class and then expose the CrUD operations and the `sync`/`reset` methods from this adapter/service/wrapper.

Reply:
Currently our thinking is that, having the repository issue HTTP requests to the internal airbyte ConfigApi server rather than talking to the underlying datasources is a short/intermediate solution to producing Public API endpoints.

I expect the public API server to slowly transition talking directly to the underlying datasources and replacing some of the endpoints in the internal config API server, leaving the internal config API server to serve more configuration or workflow internal specific functionality.

With that high level plan said, would you still recommend keeping code that doesn't talk to the underlying datasource outside of a repository for now to keep things more clear for now? Or is leaving those there with some explanatory comments satisfactory.

Reply:
By definition, the Micronaut repositories are meant to be linked to a SQL `DataSource`.  If you don't specify which data source to use, it uses the default one configured by the application.  

> The [@Repository](https://micronaut-projects.github.io/micronaut-data/latest/api/io/micronaut/data/annotation/Repository.html) annotation accepts an optional string value which represents the name of the connection or datasource in a multiple datasource scenario. By default, Micronaut Data will look for the default datasource.

Adding other methods that don't use a `DataSource` is a mixing of different concerns.  If the intent is to have one single place to go to fetch data, I would still recommend putting a wrapper class together that can delegate to either the repository or the HTTP client so that everything is kept separate and easy to test.

Reply:
Thanks for the insights! That all makes sense to me, I'll go ahead and make that wrapper class.

Reply:
I've split the two into a repository and an `internalApiWrapper`. Regarding the repository, i ran into a error while building when the `@repository` annotation was added to `ConnectionsRepository` as opposed to `ConnectionsRepositoryImpl`. The error is:
```
> Task :airbyte-api-server:compileJava FAILED
/Users/tcho/code/airbyte/airbyte-api-server/src/main/java/io/airbyte/api/server/repositories/ConnectionsRepository.java:15: error: Unable to implement Repository method: ConnectionsRepository.sync(UUID connectionId). No possible implementations found.
  void sync(UUID connectionId);
       ^
Note: Creating bean classes for 3 type elements
1 error
```

I also had to add a `@Bean` annotation to the wrapper. I'm not sure if there's a better annotation to use for that case.